packages/suite-base/src/providers/CurrentLayoutProvider/loadDefaultLayouts.ts

```diff
@@ -0,0 +1,59 @@
+// SPDX-FileCopyrightText: Copyright (C) 2023-2024 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)<lichtblick@bmwgroup.com>
+// SPDX-License-Identifier: MPL-2.0
+
+import Logger from "@lichtblick/log";
+import { LayoutLoader } from "@lichtblick/suite-base/services/ILayoutLoader";
+import { ILayoutManager } from "@lichtblick/suite-base/services/ILayoutManager";
+
+const log = Logger.getLogger(__filename);
+
+const isFulfilled = <T>(result: PromiseSettledResult<T>): result is PromiseFulfilledResult<T> =>
+  result.status === "fulfilled";
+
+const isRejected = (result: PromiseSettledResult<unknown>): result is PromiseRejectedResult =>
+  result.status === "rejected";
+
+export const loadDefaultLayouts = async (
+  layoutManager: ILayoutManager,
+  loaders: readonly LayoutLoader[],
+): Promise<void> => {
+  if (loaders.length === 0) {
+    return;
+  }
+
+  try {
+    const currentLayouts = await layoutManager.getLayouts();
+    const currentLayoutsFroms = new Set(currentLayouts.map((layout) => layout.from));
+
+    const loaderPromises = loaders.map(async (loader) => await loader.fetchLayouts());
+    const loaderResults = await Promise.allSettled(loaderPromises);
+
+    const newLayouts = loaderResults
+      .filter(isFulfilled)
+      .flatMap((result) => result.value)
+      .filter((layout) => !currentLayoutsFroms.has(layout.from));
+
+    // Log errors cause failed to fetch some layout from a specific loader
+    loaderResults.filter(isRejected).forEach((result) => {
+      log.error(`Failed to fetch layouts from loader: ${result.reason}`);
+    });
+
+    const savedPromises = newLayouts.map(
+      async (layout) =>
+        await layoutManager.saveNewLayout({
+          ...layout,
+          permission: "CREATOR_WRITE",
+        }),
+    );
+
+    // Try to save all layouts
+    const saveResults = await Promise.allSettled(savedPromises);
+
```

Comment:
renamed to savedResult
```suggestion
    const savedResults = await Promise.allSettled(savedPromises);
```